### PR TITLE
Shortcodes: add Codepen

### PR DESCRIPTION
**Pull request by jeherve**
Ported from WordPress.com

To embed a Codepen, paste a Codepen URL on a separate line, e.g.

```
http://codepen.io/css-tricks/pen/wFeaG
```

### CODE REVIEW

File: modules/shortcodes/codepen.php [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+/*
+ * CodePen embed
+ *
+ * example URL: http://codepen.io/css-tricks/pen/wFeaG
+*/
+
+// Register oEmbed provider
+wp_oembed_add_provider( '#http[s]{0,1}://codepen.io/([^/]+)/pen/([^/]+)[/]{0,1}#', 'https://codepen.io/api/oembed', true );
```
Comment (aduth) [inline modules/shortcodes/codepen.php]: Minor details, but both `[s]{0,1}` and `[/]{0,1}` could be simplified to `s?` and `/?` respectively.

**Comment by aduth** [comment]
Works well in my testing :+1:

File: modules/shortcodes/codepen.php (hunk 1 above)
Comment (jeherve) [inline modules/shortcodes/codepen.php]: @aduth Thanks! I implemented your changes in a7e12d258a2275cc759ee180a43ee24908682b94

**Comment by aduth** [comment]
Still tests well after revisions.

**Pull request merged**